src/StateSpace.hpp

```diff
@@ -23,11 +23,16 @@ class StateSpace {
      * Finds a state in the direction of @target from @source.state().
      * This new state will potentially be added to the tree.  No need to do
      * any validation on the state before returning, the tree will handle
-     * that.
+     * that. A negative stepSize tells it to implement adaptive stepsize control.
      */
     virtual T intermediateState(const T &source, const T &target, float stepSize) const = 0;
 
     /**
+     * An overloaded version designed for use in adaptive stepsize control.
```

Comment:
Can you document here what ascLimit does?  Maybe instead you could do something like:

~~~{.cpp}
typedef struct {
   float min;
   float max;
}
virtual T intermediateState(const T &source, const T &target, StepSizeLimits limits) const = 0;
~~~

Then it could be called like:

~~~{.cpp}
state = intermediateState(from, to, {2, 4})
~~~